bagel/cli.py

```diff
@@ -305,36 +305,43 @@ def bids(
                 session=session,
             )
 
-            # If subject's session has no image files, a Session object is not added
             if not image_list:
                 continue
 
             # TODO: Currently if a subject has BIDS data but no "ses-" directories (e.g., only 1 session),
-            # we create a session for that subject with a custom label "ses-nb01" to be added to the graph
-            # so the API can still find the session-level information.
+            # we create a session for that subject with a custom label "ses-nb01" to be added to the graph.
+            # However, we still provide the BIDS SUBJECT directory as the session path, instead of making up a path.
             # This should be revisited in the future as for these cases the resulting dataset object is not
             # an exact representation of what's on disk.
-            session_label = CUSTOM_SESSION_ID if session is None else session
+            # Here, we also need to add back "ses" prefix because pybids stripped it
+            session_label = (
+                CUSTOM_SESSION_ID if session is None else f"ses-{session}"
+            )
             session_path = butil.get_session_path(
                 layout=layout,
                 bids_dir=bids_dir,
                 bids_sub_id=bids_sub_id,
                 session=session,
             )
 
-            session_list.append(
-                # Add back "ses" prefix because pybids stripped it
-                models.ImagingSession(
-                    hasLabel="ses-" + session_label,
+            # If a custom Neurobagel-created session already exists (if `bagel derivatives` was run first),
+            # we add to that session when there is no session layer in the BIDS directory
+            if session_label in existing_sessions_dict:
+                existing_img_session = existing_sessions_dict.get(
+                    session_label
+                )
+                existing_img_session.hasAcquisition = image_list
+                existing_img_session.hasFilePath = session_path
+            else:
+                new_imaging_session = models.ImagingSession(
+                    hasLabel=session_label,
                     hasFilePath=session_path,
                     hasAcquisition=image_list,
                 )
-            )
-
-        pheno_subject.hasSession += session_list
+                existing_subject.hasSession.append(new_imaging_session)
 
     context = generate_context()
-    merged_dataset = {**context, **pheno_dataset.dict(exclude_none=True)}
+    merged_dataset = {**context, **jsonld_dataset.dict(exclude_none=True)}
```

Comment:
My IDE says about this line:

```
The method "dict" in class "BaseModel" is deprecated
  The `dict` method is deprecated; use `model_dump` instead.Pylance
```
Mind taking a look?

Reply:
`model_dump` is only available with Pydantic v2! Would need to be addressed in https://github.com/neurobagel/bagel-cli/issues/180